### PR TITLE
Ds/more binding goodness

### PR DESCRIPTION
**Pull request by dsteinmo**
* Note: Ignore appveyor failures, still unmerged/broken.

### CODE REVIEW

File: include/BlitzHelpers.hpp [hunk 1]
```diff
@@ -190,4 +318,116 @@ namespace blitzdg {
...
+            scatter[seed] = nunique++;
+            gather.erase(first, last);
+            ++curr;
+        }
+        return std::make_pair(std::move(gather), std::move(scatter));
+    }
+
+    // Order the columns of A by decreasing size of their 2-norm.
+    template <typename T>
+    std::vector<index_type> orderColsByNorm(const matrix_type<T>& A) {
```
Comment (k7miller) [inline include/BlitzHelpers.hpp:387]: This function should go into the details namespace.

File: include/BlitzHelpers.hpp [hunk 2]
```diff
@@ -190,4 +318,116 @@ namespace blitzdg {
...
+        using details::CompareEQByDim;
+        using details::CompareExact;
+        using details::getOrdering;
+
+        if (tol <= real_type(0)) {
+            return unique(A);
+        }
+        std::vector<index_type> gather(A.rows()), scatter(A.rows());
+        std::iota(gather.begin(), gather.end(), index_type(0));
+        auto order = getOrdering(A);
```
Comment (k7miller) [inline include/BlitzHelpers.hpp:358]: I was lazy with my naming here. I would change `getOrdering` to `orderColsByVariance`.

File: include/DGContext2D.hpp [hunk 3]
```diff
@@ -22,6 +22,8 @@ namespace blitzdg {
         real_matrix_type* yGrid_;
         real_matrix_type* Fscale_;
         index_matrix_type* Fmask_;
+        std::vector<index_type>* Gather_;
```
Comment (k7miller) [inline include/DGContext2D.hpp:25]: I assume you are storing everything here by raw pointer since `DGContext2D` is non-owning of these resources?

**Review by k7miller — APPROVED** [review]
Looks good buddy. Just a few minor comments. Nice job! :100:

File: include/DGContext2D.hpp (hunk 3 above)
Comment (dsteinmo) [reply include/DGContext2D.hpp:25]: Yessir

**Pull request merged**